core/dbt/adapters/sql/impl.py

```diff
@@ -172,15 +172,19 @@ def create_schema(self, database, schema):
         }
         self.execute_macro(CREATE_SCHEMA_MACRO_NAME, kwargs=kwargs)
         self.commit_if_has_connection()
+        # we can't update the cache here, as if the schema already existed we
+        # don't want to (incorrectly) say that it's empty
 
     def drop_schema(self, database, schema):
         logger.debug('Dropping schema "{}"."{}".', database, schema)
         kwargs = {
             'database_name': self.quote_as_configured(database, 'database'),
             'schema_name': self.quote_as_configured(schema, 'schema'),
         }
-        self.execute_macro(DROP_SCHEMA_MACRO_NAME,
-                           kwargs=kwargs)
+        self.execute_macro(DROP_SCHEMA_MACRO_NAME, kwargs=kwargs)
+        self.commit_if_has_connection()
```

Comment:
Do we want to commit here? I'm not sure a consumer of this method would expect a commit to occur, and I can imagine this being dangerous for certain uses!

Reply:
I thought we did because the `create schema` also commits. I sort of assumed they both should, I'll revert it.

Reply:
hmm, `create_schema` should probably not commit - these sprinkled-in transactional statements make it super hard to reason about what dbt is doing in different parts of the codebase.


I do think that committing a `create schema` feels "safer" than committing the drop - it's the destructive nature of this method that gave me pause. Reverting it sounds good to me -- ship it when the tests pass!